docs/jwst/user_documentation/running_pipeline_python.rst

```diff
@@ -0,0 +1,528 @@
+be saved (so, also setting ``save_results`` is redundant), and that a new file
+base name should be used with the appropriate file suffix appended. For example,
+to save the intermediate result from the saturation step when running
+``Detector1Pipeline`` with a file name based on the string `saturation_result` instead
+of `jw00017001001_01101_00001_nrca1`:
+
+::
+
+	# To save the intermediate results of a step within a pipeline to a file with a custom name
+	 from jwst.pipeline import Detector1Pipeline
+	 result = Detector1Pipeline.call('jw00017001001_01101_00001_nrca1_uncal.fits',
+	 								steps={"saturation": {"output_file": 'saturation_result'})
+
+Similarly, when `output_file` is set on an individual step class, this will indicate
+that the result from that step should be saved to a file with that basename and the
+appropriate suffix.
+
+::
+
+	# To save the final results from SaturationStep with a custom output file name when run standalone
+	 from jwst.linearity import SaturationStep
+	 SaturationStep.call('jw00017001001_01101_00001_nrca1_uncal.fits', output_file="saturation_result")
+
+**Setting Output File Directory** 
+
+Setting ``output_dir`` at the step-level indicates that the step's result should
+be saved (so, also setting ``save_results`` is redundant), and that the files
+should be saved in the directory specified instead of the current working directory.
+For example, to save the intermediate results of ``DarkCurrentStep`` when running
+``Detector1Pipeline`` in a subdirectory ``/calibrated``:
+
+::
+
+	# to save the intermediate step result in a different specified output directory
+	 from jwst.pipeline import Detector1Pipeline
+	 result = Detector1Pipeline.call('jw00017001001_01101_00001_nrca1_uncal.fits',
+	 								 steps={'dark': {'output_dir': 'calibrated'}})
+
+
+Similarly, when `output_dir` is set on an individual step class, this will indicate
+that the result from that step should be saved to the specified directory:
+
+::
+
+	# to save the final result of a 
+	 from jwst.pipeline import Detector1Pipeline
+	 result = DarkCurrentStep.call('jw00017001001_01101_00001_nrca1_uncal.fits', output_dir='calibrated')
+
+
+.. _python_run_vs_call:
+
+Advanced use - `pipeline.run()` vs. `pipeline.call`
+===================================================
+
+Another option for running the pipeline is to use the `.run()` method instead
+of the `.call()` method. This method is not generally reccomended and considered
+advanced use, but it is an option to users.
+
+There is additional internal setup done when using the ``.call()`` method to ensure
+that reference files and parameter files from CRDS are passed to the pipeline along
+with any other parameters passed directly to the method. In the process of doing this,
+a new instance of the pipeline/step class is created internally, to which the final
+set of parameters is passed to and the ``.run()`` method, which is what finally begins
```

Comment:
I'm not sure how to parse this whole sentence. Are you saying that the way it works *internally* when you use .call() is that a new instance is created, parameters are retrieved, and then a call is made to .run() using those parameters? If so, I think the sentence needs a little rearranging to make that more obvious.

Reply:
yeah thats what i was trying to describe, ill reword it.